forward_engineering/script_parser/curl_script_parser.js

```diff
@@ -0,0 +1,153 @@
+const curlPostRegex = /-X\s*[Pp][Oo][Ss][Tt]/g;
+const curlPutRegex = /-X\s*[Pp][Uu][Tt]/g;
+
+/**
+ * @param firstLine {string}
+ * @throws {Error}
+ * @return string
+ */
+const extractHttpMethod = (firstLine) => {
+    const isPutRequest = curlPutRegex.test(firstLine);
+    if (isPutRequest) {
+        return 'PUT';
+    }
+    const isPostRequest = curlPostRegex.test(firstLine);
+    if (isPostRequest) {
+        return 'POST';
+    }
+    throw new Error(`Invalid curl config: ${firstLine}`);
+}
+
+/**
+ * @param firstLine {string}
+ * @param httpMethod {string}
+ * @throws {Error}
+ * @return string
+ */
+const extractIndexName = (firstLine, httpMethod) => {
```

Comment:
Is that a good way to get index name? It seems better to parse json data from entitiesData and get inedx name from there.

Reply:
JSON data is not always there